methods/guest/src/main.rs

```diff
@@ -1,24 +1,58 @@
 use risc0_zkvm::guest::env;
 
-use consensus::beacon_state;
-use consensus::beacon_block::BeaconBlock;
-use consensus::beacon_block_header::BeaconBlockHeader;
+use alloy_primitives::FixedBytes;
+use consensus_common::{
+    beacon_block::BeaconBlock,
+    beacon_block_header::BeaconBlockHeader,
+    funcs::{process_block_header, ProcessBlockHeaderOutput},
+    proof::Proof,
+};
+use ssz_rs::prelude::*;
 
 fn main() {
     // let count_start = env::cycle_count();
 
+    let pre_state_root: FixedBytes<32> = env::read();
+
     let slot: u64 = env::read();
-    let latest_block_header: BeaconBlockHeader = env::read();
+    let slot_proof: Proof = env::read();
+    let latest_block_header_ssz: Vec<u8> = env::read();
+    let latest_block_header: BeaconBlockHeader =
+        BeaconBlockHeader::deserialize(&mut latest_block_header_ssz.as_slice()).unwrap();
+    let latest_block_header_proof: Proof = env::read();
+    // let validator_slashed: bool = env::read();
+    // let validator_slashed_proof: Proof = env::read();
+
     let proposer_index: u64 = env::read();
```

Comment:
missing verification for `proposer_index`

Reply:
I added TODO that this should be implemented in the future